airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/base/DestinationRunner.java

```diff
@@ -0,0 +1,85 @@
+import io.airbyte.commons.io.IOs;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.protocol.models.v0.AirbyteMessage;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog;
+import io.airbyte.validation.json.JsonSchemaValidator;
+import java.nio.file.Path;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+public class DestinationRunner extends IntegrationRunner<AirbyteDestinationConfig> {
+  private final Destination destination;
+
+  public DestinationRunner(final Destination destination) {
+    super(new IntegrationCliParser(), Destination::defaultOutputRecordCollector);
+    this.destination = destination;
+  }
+
+  @VisibleForTesting
+  DestinationRunner(final IntegrationCliParser cliParser,
+      final Consumer<AirbyteMessage> outputRecordCollector,
+      final Destination destination) {
+    super(cliParser, outputRecordCollector);
+    this.destination = destination;
+  }
+
+  @VisibleForTesting
+  DestinationRunner(final IntegrationCliParser cliParser,
+      final Consumer<AirbyteMessage> outputRecordCollector,
+      final Destination destination,
+      final JsonSchemaValidator jsonSchemaValidator) {
+    super(cliParser, outputRecordCollector, jsonSchemaValidator);
+    this.destination = destination;
+  }
+
+  @Override
+  protected Integration getIntegration() {
+    return destination;
+  }
+
+  @Override
+  protected void check(IntegrationConfig parsed) throws Exception {
+    final AirbyteDestinationConfig config = parseConfig(parsed.getConfigPath());
+    DestinationConfig.initialize(config);
+    check(config);
+  }
+
+  @Override
+  protected void discover(IntegrationConfig parsed) throws Exception {
+    throw new IllegalStateException("Cannot execute discover on a destination!");
+  }
+
+  protected void read(IntegrationConfig parsed) throws Exception {
+    throw new IllegalStateException("Cannot execute read on a destination!");
+  }
+
+  @Override
+  protected void write(final IntegrationConfig parsed) throws Exception {
+    final AirbyteDestinationConfig config = parseConfig(parsed.getConfigPath());
+    validateConfig(destination.spec().getConnectionSpecification(), config, "WRITE");
+    // save config to singleton
+    DestinationConfig.initialize(config);
+    final ConfiguredAirbyteCatalog catalog = Jsons.object(Jsons.deserialize(IOs.readFile(parsed.getCatalogPath())), ConfiguredAirbyteCatalog.class);
+
+    try (final SerializedAirbyteMessageConsumer consumer = destination.getSerializedMessageConsumer(config, catalog, outputRecordCollector)) {
+      consumeWriteStream(consumer);
+    } finally {
+      stopOrphanedThreads(EXIT_HOOK,
+          INTERRUPT_THREAD_DELAY_MINUTES,
+          TimeUnit.MINUTES,
+          EXIT_THREAD_DELAY_MINUTES,
+          TimeUnit.MINUTES);
+    }
+  }
+
+  @Override
+  public AirbyteDestinationConfig parseConfig(Path path) {
+    return AirbyteDestinationConfig.fromPath(path);
+  }
+}
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: lines 86 to 86]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->Wasn't this file created in an earlier PR in this stack?<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/33559?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->